### PR TITLE
Backport v0.18 (15273, 15337, 15297, 15322, 15196, 14884, 14966, 15012, 14319, 14809, 14783, 14820, 14678)

### PR DESCRIPTION
**Pull request by linuxsh2**
A few cherry-picks that had no conflicts.

### CODE REVIEW

**Comment by PastaPastaPasta** [comment]
We don't want to backport 11526 unless someone will modify it to actually work with Dash buidling

I think appveyor depends on 11526, so can't have that either

**Comment by linuxsh2** [comment]
Should I mark 11526 as DNM then?

**Comment by PastaPastaPasta** [comment]
Probably not... I added a comment outlining the situation

File: doc/README.md [hunk 1]
```diff
@@ -36,9 +36,10 @@ The following are developer notes on how to build Dash Core on your native platf
 - [macOS Build Notes](build-osx.md)
 - [Unix Build Notes](build-unix.md)
 - [Windows Build Notes](build-windows.md)
+- [FreeBSD Build Notes](build-freebsd.md)
 - [OpenBSD Build Notes](build-openbsd.md)
 - [NetBSD Build Notes](build-netbsd.md)
-- [Gitian Building Guide](gitian-building.md)
+- [Gitian Building Guide (External Link)](https://github.com/bitcoin-core/docs/blob/master/gitian-building.md)
```
Comment (UdjinM6) [inline doc/README.md:42]: 15132: none of this is relevant, should skip it

File: test/lint/check-doc.py [hunk 2]
```diff
@@ -26,8 +26,12 @@
 
 
 def main():
-    used = check_output(CMD_GREP_ARGS, shell=True, universal_newlines=True, encoding='utf8')
-    docd = check_output(CMD_GREP_DOCS, shell=True, universal_newlines=True, encoding='utf8')
+    if sys.version_info >= (3, 6):
+        used = check_output(CMD_GREP_ARGS, shell=True, universal_newlines=True, encoding='utf8')
+        docd = check_output(CMD_GREP_DOCS, shell=True, universal_newlines=True, encoding='utf8')
```
Comment (UdjinM6) [inline test/lint/check-doc.py:31]: 15196: This part is from 14884, should backport it first

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

**Review by UdjinM6 — APPROVED** [review]
LGTM, utACK

**Review by PastaPastaPasta — APPROVED** [review]
utACK for merging via merge commit

**Pull request merged**